include/nlohmann/json.hpp

```diff
@@ -3053,12 +3046,12 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         iterator result(this);
         JSON_ASSERT(m_value.array != nullptr);
 
-        auto insert_pos = std::distance(m_value.array->begin(), pos.m_it.array_iterator);
-        m_value.array->insert(pos.m_it.array_iterator, std::forward<Args>(args)...);
-        result.m_it.array_iterator = m_value.array->begin() + insert_pos;
+        auto insert_pos = std::distance(m_value.array->begin(), pos.m_it.array_iterator());
+        m_value.array->insert(pos.m_it.array_iterator(), std::forward<Args>(args)...);
+        result.m_it.set_array_iterator(m_value.array->begin() + insert_pos);
 
         // This could have been written as:
-        // result.m_it.array_iterator = m_value.array->insert(pos.m_it.array_iterator, cnt, val);
+        // result.m_it.set_array_iterator(m_value.array->insert(pos.m_it.array_iterator(), cnt, val));
```

Comment:
I've come across a related Codacy issue today ... both things could be solved with a helper. Doesn't have to be here though.